### DISCUSSION TITLE
BepInEx with Yooka-Laylee (Unity 5 Game)

### DISCUSSION THREAD

**Discussion by SpringEThing** [BepInEx/BepInEx#377] [Help and Q&A] [answered]
Hi guys, encountered a small annoyance with Yooka-Laylee (otherwise enjoying the game a lot) that I'm really sure I could make some Harmony patches for in like a night, but I can't figure out the right BepInEx setup to get a plugin I build to do anything. I have good experience modding Nick All-Star Brawl (NASB) using BepInEx and so I was able to translate my knowledge for setup for that game here, or so I thought.

Using BepInEx 5 on the game just gives a black screen on boot, so I had to go to the next best thing and use the very last release of BepInEx 4, that being release 4.1.2, this at least got the game to boot properly although I didn't try actually playing the game.
The .NET framework of 'Assembly-CSharp.dll' for the game is 'v2.0.50727', the earliest .NET framework my VS project was able to target when referencing DLLs from BepInEx 4.1.2 was .NET Framework 3.5, so I went with that. Trying .NET Framework 2 made the DLL refuse to build due to some loose dependencies in my reference DLLs. Anyway, with framework 3.5 The plugin built, and the entire source code is as follows:
```
using System;
using System.Collections.Generic;
using System.Text;
using BepInEx;
using Harmony;
using UnityEngine;

namespace ImprovedHunterTonic
{
    [BepInPlugin(pluginGuid,pluginName,pluginVersion)]
    public class ImprovedHunterTonic : BaseUnityPlugin
    {
        public const string pluginGuid = "jarmstrongart.yookalaylee.improvedhuntertonic";
        public const string pluginName = "Improved 'Hunter' Tonic";
        public const string pluginVersion = "1.0.0.0";

        public void Awake()
        {
            BepInEx.Logger.Log(BepInEx.Logging.LogLevel.Message, "it doesnt work : (");

        }
    }
}

```

As you can tell by my test log message, it doesn't work, ironic since I'd only see that log in the console if it DOES work.

I tried a few things from here:
1. Made sure to enable the Message LogLevel in the BepInEx config.
2. Tried placing the plugin in both 'patches' and 'plugins' between separate game boots

But no luck!
When the plugin was in the 'patches' folder, it at least recognised the plugin and said only '0 patches loaded from <plugin name here>', but when it was in the 'plugins' folder it acted like the plugin didn't even exist, and I assume this is because BepInEx didn't have a plugins folder or something.

Any help guys? I think the best result would be if I could just get BepInEx 5 to boot the game, then I don't think there would be any issues since I could use a similar BepInEx version to when I modded NASB.

**Comment by ghorsington** [comment, marked as answer]
Greetings!

When it comes to anything pre-2017 Unity, it might help trying some options in the [troubleshooting guide](https://docs.bepinex.dev/articles/user_guide/troubleshooting.html#unity-5-and-older). Specifically, I'd try to change the entrypoint in the configuration file in the hopes that it's just a timing issue.

Other than that, are there any logs with BepInEx 5 (e.g. preloader error logs that will appear in the game folder if there are errors)?

**Reply by SpringEThing** [reply]
> 

Thanks for getting back so quick!
I have BepInEx 4 installed right now so I'll have to change back to BepInEx 5 and rebuild my plugin, but yeah I'll try changing the entry point for sure, as one thing to note is that Yooka-Laylee does an odd thing where it restarts the game as soon as it boots and then actually boots for real, so changing the entry point might be just what I need.
Also no, there are no logs, I was going to mention this in my post but I forgot, BepInEx 4 works perfectly and causes no log generation, it also just makes it so that my plugin does nothing.

**Reply by SpringEThing** [reply]
> 

Fantastic news, changing the entry point to 'Camera' as shown in the example actually worked perfectly, and I was only doing that to see what effect it'd have if any! The game now boots with BepInEx 5 AND my plugin rebuilt for BepInEx5 works like a charm too, thank you so much!
